src/components/AbstractSelectableComponent.js

```diff
@@ -29,107 +26,43 @@ export default class AbstractSelectableComponent extends React.Component {
-      // If you want to update selection during drag, un-comment this:
-      //   this.onUpdateSelection(event);
-      // See: https://github.com/hms-dbmi/vitessce/issues/111
-    }
+  getCellCoords() { // eslint-disable-line class-methods-use-this
+    // No-op
   }
 
-  onDragEnd(event) {
-    const { tool } = this.state;
-    if (tool === SELECT_RECTANGLE) {
-      this.setState({ selectionRectangle: undefined });
-      this.onUpdateSelection(event);
-    }
+  getCellBaseLayerId() { // eslint-disable-line class-methods-use-this
+    // No-op
   }
 
-  onUpdateSelection(event) {
+  renderSelectionLayers() {
     const { tool } = this.state;
     const { updateCellsSelection } = this.props;
-    if (tool === SELECT_RECTANGLE && event.coordinate && updateCellsSelection) {
-      const {
-        xMin, yMin, xMax, yMax,
-      } = this.getDragRectangle(event);
-      // The built-in pickObjects is limited in the size of the region that can be selected.
-      // https://github.com/uber/deck.gl/issues/2658#issuecomment-463293063
-      // (and it would also be limited in resolution of features it can pick.)
-
-      // I've had other problems as well: Either not working at all,
-      // or only picking objects with positive coordinates.
-
-      const selectedCellIdsSet = {};
-
-      const { viewport } = this.viewInfo;
-
-      const nwCoords = viewport.project([xMin, yMin]);
-      const seCoords = viewport.project([xMax, yMax]);
-      const proWidth = seCoords[0] - nwCoords[0];
-      const proHeight = seCoords[1] - nwCoords[1];
-      const projectedProps = {
-        x: nwCoords[0],
-        y: nwCoords[1],
-        width: proWidth,
-        height: proHeight,
-      };
-      const pickedObjects = this.deckRef.current.pickObjects(projectedProps);
-
-      const cellObjIds = pickedObjects.map(cellObj => cellObj.object[0]);
-      cellObjIds.forEach((cellObjId) => {
-        selectedCellIdsSet[cellObjId] = true;
-      });
-      updateCellsSelection(selectedCellIdsSet);
-    }
-  }
-
-  renderSelectionRectangleLayers() {
-    const { selectionRectangle } = this.state;
-    if (!selectionRectangle || !this.dragStartCoordinate) {
+    if (!tool) {
       return [];
     }
-    return [new PolygonLayer({
+    return [new SelectionLayer({
+      id: 'selection',
+      getCellCoords: this.getCellCoords,
       coordinateSystem: COORDINATE_SYSTEM.IDENTITY,
-      id: 'selection-rectangle',
-      data: [selectionRectangle],
-      getPolygon(bounds) {
-        return [
-          [bounds.xMin, bounds.yMin],
-          [bounds.xMax, bounds.yMin],
-          [bounds.xMax, bounds.yMax],
-          [bounds.xMin, bounds.yMax],
-        ];
+      selectionType: tool,
+      onSelect: ({ pickingInfos }) => {
+        const cellObjIds = pickingInfos.map(cellObj => cellObj.object[0]);
+        const selectedCellIdsSet = {};
+        cellObjIds.forEach((cellObjId) => {
+          selectedCellIdsSet[cellObjId] = true;
+        });
```

Comment:
Maybe not for this PR, but is there any reason not to use a real Set object here?

Reply:
I can try with a Set object. I just basically was doing the same thing that was being done with the previous rectangle selection code https://github.com/hms-dbmi/vitessce/pull/254/files#diff-a42ebb9da25dd2ce5723a87c2f65de8bL103

Reply:
I have updated to use `Set` and everything seems to work well.